### ISSUE TITLE
Excluding known typed fields from returned map

### ISSUE THREAD

**Issue by thedenisnikulin** [PerimeterX/marshmallow#16]
It would be great if there was an option that excludes from resulting `map[string]interface{}` the fields that were unmarshalled to the struct object, see the example below:

```golang
v := struct {
	Foo string `json:"foo"`
	Boo []int  `json:"boo"`
}{}
result, err := marshmallow.Unmarshal([]byte(`{"foo":"bar","boo":[1,2,3],"goo":12.6}`), &v)
fmt.Printf("v=%+v, result=%+v, err=%v", v, result, err)
// Output: v={Foo:bar Boo:[1 2 3]}, result=map[goo:12.6], err=<nil>
//                                  ^^^^^^     ^^^
// only "goo" field is present in result since "foo" and "boo" are explicitly typed
```

**Comment by avivpxi** [comment]
Yes, @thedenisnikulin, it seems reasonable and should be quite easy to add.

We will be able to get to it in the upcoming weeks.
At the meantime, if you're interested in submitting a PR yourself, that would be appreciated

**Comment by thedenisnikulin** [comment]
Great, thanks!

**Comment by avivpxi** [comment]
@thedenisnikulin support for excluding known fields is released in v1.1.3
Example can be found [here](https://github.com/PerimeterX/marshmallow/blob/main/example_test.go#L59).
Thanks @galepx and @itayPX for moving fast with this one 👑 🙏
